### ISSUE TITLE
When an extension that other extensions depend on is disabled, all the extensions that depend on it should automatically be disabled

### ISSUE THREAD

**Issue by alexr00** [Microsoft/vscode#59500]
The built-in "Node Debug (legacy)" extension depends on "Node Debug".

- Try to disable "Node Debug" but nothing happens, presumably because it has a dependency
- Try to disable "Node Debug (legacy)" and it works
- Then, also disable "Node Debug" and now it works, presumably because the dependency is gone
- Finally, try to enable "Node Debug (legacy)" and it works?! and also automatically enabled "Node Debug"

To be consistent, if it is possible to enable "Node Debug (legacy)" when "Node Debug" is disabled and automatically enable "Node Debug" then it should also be possible to disable "Node Debug" and have "Node Debug (legacy)" also be automatically disabled.

This is also confusing because there's no indication why nothing happened when I clicked on Disable on "Node Debug" while "Node Debug (legacy)" is enabled.

**Comment by sandy081** [comment]
> Try to disable "Node Debug" but nothing happens, presumably because it has a dependency

You see an error at the bottom right that there is an extension (Node Debug Legacy) depends on it

> Try to disable "Node Debug (legacy)" and it works

👍 

> Then, also disable "Node Debug" and now it works, presumably because the dependency is gone

👍 

> Finally, try to enable "Node Debug (legacy)" and it works?! and also automatically enabled "Node Debug"

Yes, enabling/installing an extension also installs its dependency. If a user is trying to enable/install an extension, it is implied dependencies too.

> To be consistent, if it is possible to enable "Node Debug (legacy)" when "Node Debug" is disabled and automatically enable "Node Debug" then it should also be possible to disable "Node Debug" and have "Node Debug (legacy)" also be automatically disabled.

Other way around is not good. Uninstalling / Disabling dependencies of an extension together might not be expected and users would have installed them independently. It was this behaviour (with prompt) but I should say it was not the best. Hence made disablement and uninstalling independent.

> This is also confusing because there's no indication why nothing happened when I clicked on Disable on "Node Debug" while "Node Debug (legacy)" is enabled.

There is. As I said you see error prompt at the bottom right

![image](https://user-images.githubusercontent.com/10746682/46146198-7c223400-c262-11e8-8b0f-f330a68af554.png)

**Comment by alexr00** [comment]
I don't get the error prompt at all :(

**Comment by sandy081** [comment]
Strange I see it. May be try with all extensions disabled?

![kapture 2018-09-27 at 23 13 19](https://user-images.githubusercontent.com/10746682/46175054-f9719700-c2aa-11e8-8da1-740ea4a65e8a.gif)

**Comment by alexr00** [comment]
I disabled all extensions then restarted VS Code. Still nothing ☹️ What other info can I collect for you?

**Comment by alexr00** [comment]
Node Debug and Node Debug (legacy) don't appear to be installed when I run from dev (e0e2738819c0f) I'm looking for another pair to try with. Let me know if you know of one or if there's a good way to check extension dependency.

**Comment by alexr00** [comment]
Tried with the GitHub Pull Requests extension and the built-in Git extension. Same results: when I try to disable the Git extension it does not disable (👍) but there is no notification(👎).

**Comment by sandy081** [comment]
@alexr00 Can you please breakpoint at the following line and see if it is getting hit?

https://github.com/Microsoft/vscode/blob/a144e2fa9fc2fb91a8add05833943a00f3b86371/src/vs/workbench/parts/extensions/node/extensionsWorkbenchService.ts#L775

If it is, can you please check why the error is not being notified?

**Comment by alexr00** [comment]
@sandy081, the error tries to fire, but in event.ts on line 132 `this._listeners` is undefined. `this._deliveryQueue` is also undefined at the start of the fire function.

**Comment by sandy081** [comment]
@bpasero 

For @alexr00 When an action is executed and returns an error, it is not notified. 

https://github.com/Microsoft/vscode/blob/4b427941dc20b303e35c7687924a4a92f73c4c20/src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts#L729

Any insights?

**Comment by bpasero** [comment]
> Any insights?

Any steps?

**Comment by sandy081** [comment]
Try to disable an extension which is a dependent of some other extension. You can try disabling `Node Debug` or install `Pull Request Extension` and disable `Git`. You should see an error notification.

**Comment by bpasero** [comment]
@sbatten I can reproduce and it looks like neither our custom context menu nor our native context menu delegate failing actions to a notification service to show the error. I think we should probably hook that in. I think this is not a recent regression, this probably never worked properly (or at least for a while).

**Comment by bpasero** [comment]
Maybe we should open a separate issue for the menu issue, up to you @sandy081

**Comment by sandy081** [comment]
I think its fine to use this issue and update the summary or description if needed.

**Comment by sbatten** [comment]
I don't know when or how this was fixed, but I can not reproduce any longer. @bpasero @alexr00 to verify?

**Comment by alexr00** [comment]
I can still reproduce by trying to disable Node Debug.

**Comment by sbatten** [comment]
![image](https://user-images.githubusercontent.com/6561887/49962714-d8dcd480-feca-11e8-86f3-4ce1dff5871f.png)

**Comment by alexr00** [comment]
Correct. I don't see that prompt. It looks as though nothing happens when I try to disable Node Debug.

**Comment by sbatten** [comment]
@alexr00 do you see the error in the devtools console?

**Comment by alexr00** [comment]
No, I don't see the error in the devtools console. I just get this when I right click:

dom.ts:213 [Violation] Added non-passive event listener to a scroll-blocking 'mousewheel' event. Consider marking event handler as 'passive' to make the page more responsive. See https://www.chromestatus.com/feature/5745543795965952

**Comment by sbatten** [comment]
@alexr00 oh im stupid, I was using the gear menu, not the context menu to disable. I can repro

**Comment by sbatten** [comment]
@alexr00 this seems to be resolved for the context menu now, tested disabling `git` Hopefully I did not misread again, but please try it out.
